tools/linter/clang_tidy/__main__.py

```diff
@@ -173,17 +180,16 @@ def main() -> None:
     options = parse_args()
 
     # Check if clang-tidy executable exists
-    exists = os.access(options.clang_tidy_exe, os.X_OK) or shutil.which(
-        options.clang_tidy_exe
-    )
+    exists = os.access(options.clang_tidy_exe, os.X_OK)
+
     if not exists:
         msg = (
             "Could not find 'clang-tidy' binary\n"
             "You can install it by running:\n"
             "   python3 tools/linter/install/clang_tidy.py"
         )
         print(msg)
-        exit(1)
+        raise RuntimeError("Could not find clang-tidy binary")
```

Comment:
nit: why raise 2 errors?
```suggestion
        raise RuntimeError(msg)
```